src/test/java/io/github/oliviercailloux/y2018/apartments/utils/JsonConvertTest.java

```diff
@@ -83,34 +60,23 @@ void apartmentsToJsonTest() throws IOException, URISyntaxException {
 
 		String expectedApartment = "[{\"address\":\"118 rue du père noel 77480\",\"description\":\"\",\"floorArea\":1182118.48,\"floorAreaTerrace\":0.0,\"nbBathrooms\":0,\"nbBedrooms\":0,\"nbMinNight\":0,\"nbSleeping\":0,\"pricePerNight\":0.0,\"tele\":false,\"terrace\":false,\"title\":\"Grand Igloo\",\"wifi\":false},{\"address\":\"123 rue du soleil\",\"description\":\"\",\"floorArea\":1234567.89,\"floorAreaTerrace\":0.0,\"nbBathrooms\":0,\"nbBedrooms\":0,\"nbMinNight\":0,\"nbSleeping\":0,\"pricePerNight\":0.0,\"tele\":false,\"terrace\":false,\"title\":\"Maison Test\",\"wifi\":false}]";
 
-		URI ressource = JsonConvertTest.class.getResource("jsonListTest.json").toURI();
+		URI ressource = JsonConvertTest.class.getResource("jsonApartments.json").toURI();
 		Path jsonPath = Path.of(ressource);
 
 		JsonConvert.apartmentsToJson(apartments, jsonPath);
 
 		assertEquals(expectedApartment, Files.readString(jsonPath));
 		assertThrows(IOException.class, () -> JsonConvert.apartmentsToJson(apartments, Paths.get("")));
-	}
 
-	/**
-	 * Tests jsonToApartment function. Verifies if the Apartment created by the
-	 * function corresponds to the expected Apartment.
-	 *
-	 * @throws FileNotFoundException if the file doesn't exists.
-	 * @throws IOException           if the file can't be convert into JSON format.
-	 */
-	@Test
-	void jsonToApartmentTest() throws IOException, URISyntaxException {
-		Builder apartBuilder = new Apartment.Builder();
-		Apartment apartmentRef = apartBuilder.setAddress("118 rue du père noel 77480").setFloorArea(1182118.48)
-				.setTitle("Grand Igloo").setTerrace(false).setWifi(false).setTele(false).build();
-
-		URI ressource = JsonConvertTest.class.getResource("jsonfileTest.json").toURI();
-		Path jsonPath = Path.of(ressource);
+		Path defaultPath = JsonConvert.apartmentsToJson(apartments);
+		assertEquals(expectedApartment, Files.readString(defaultPath));
 
-		Apartment apartmentTest = JsonConvert.jsonToApartment(jsonPath);
+		Path differentDefaultPath = JsonConvert.apartmentsToJson(apartments);
+		assertEquals(expectedApartment, Files.readString(differentDefaultPath));
+		assertNotEquals(differentDefaultPath, defaultPath);
 
-		assertEquals(apartmentRef.hashCode(), apartmentTest.hashCode());
+		Files.delete(defaultPath);
+		Files.delete(differentDefaultPath);
```

Comment:
BP ressources

Reply:
Fix in commit 79a20a634d95f956f81cdae25561e0526b354f54